exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/FakeTimeProvider.java

```diff
@@ -16,25 +16,59 @@
 
 package com.exonum.binding.testkit;
 
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.time.ZoneOffset;
 import java.time.ZonedDateTime;
+import java.time.temporal.TemporalAmount;
 
-// TODO: update Javadocs in P2 [ECR-3051]
+/**
+ * Fake time provider for service testing. Allows to manually manipulate time that is returned
+ * by TestKit time service.
+ */
 public class FakeTimeProvider implements TimeProvider {
 
-  public FakeTimeProvider create(ZonedDateTime time) {
-    throw new UnsupportedOperationException();
+  private ZonedDateTime time;
```

Comment:
I think this class must be thread-safe (more precisely, must ensure that writes made in Java are visible by whatever native thread executes transactions of Time Oracle).

We can either put `volatile` here (with an explanation of _why_ that's enough for our use-case) or make all the operations synchronized (also with an explanation :upside_down_face: )